### DISCUSSION TITLE
State of optimization

### DISCUSSION THREAD

**Discussion by jjezewski** [anyproto/contributors#27] [Q&A]
I like to push my hardware to its limits, so for resource-intensive tasks I try to eliminate unnecessary things rather than run them in the background. Anytype is a great tool and it would be very handy to have it always on, but my concern with it is that it uses over half a gigabyte of RAM at idle. I know the desktop application is based on Electron, which leaves a significant memory footprint, but you can't really eliminate it. What gets my attention is anytype-heart (the executable is called anytypeHelper) which uses over 250MB of memory. I've done some initial code review, but before I go any further I'd like to ask the team if optimization is a priority at this stage of development, and if the resource usage currently scales well with more data (I don't have many objects in my space to test it).

**Comment by requilence** [comment]
Hey @Update450! Thank you for your feedback. We're certainly keen on optimizing memory consumption, especially since the same library is also used on mobile platforms where memory efficiency is crucial. 
We are looking into integrating benchmarks and stress tests into our development process to monitor changes between releases. 
And we would appreciate the help of our community! Are you familiar with the go profiling tools?

**Reply by jjezewski** [reply]
@requilence Not really, especially at this scale.
